File: PyhtonPractice/AbstractClassConst.py
from abc import ABC, abstractmethod

class Animal(ABC):
    
    def __init__(self, noofLeg):
        self.noofLeg = noofLeg
        
    @abstractmethod
    def makeSound(self):
        pass
        
    def sleep(self):
        print ('Zzz')
        
    def legCount(self):
        print(self.noofLeg)


class Dog(Animal):

    def __init__(self, noofLeg):
        super().__init__(noofLeg)
        
    def makeSound(self):
        print ('Bark!!!')
        
class Duck(Animal):

    def __init__(self, noofLeg):
        super().__init__(noofLeg)
        
    def makeSound(self):
        print ('Pack!!!')
        

dog = Dog(4)
dog.makeSound()
dog.sleep()
dog.legCount()
dog = Duck(2)
dog.makeSound()
dog.sleep()
dog.legCount()

File: PyhtonPractice/AbstractClassConst.java
abstract class Animal {
 private int noofLeg;
 public abstract void makeSound();
 
 public Animal(int noofLeg){
  this.noofLeg = noofLeg;
 }
 public void sleep(){
   System.out.println("Zzz");
 }
 public void legCount(){
   System.out.println(noofLeg);
 }
}

class Dog extends Animal {
  public Dog(int noofLeg) {
  super(noofLeg);
  }
  public void makeSound(){
    System.out.println("Bark!!!");
  }
}

class Duck extends Animal {
  public Duck(int noofLeg) {
  super(noofLeg);
  }
  public void makeSound(){
    System.out.println("Pack!!!");
  }
}

public class Main {
  public static void main(String[] args) {
    //System.out.println("Hello World");
    Animal dog = new Dog(4);
    dog.makeSound();
    dog.sleep();
    dog.legCount();
    Animal duck = new Duck(2);
    duck.makeSound();
    duck.sleep();
    duck.legCount();
  }
}